drivers/SmartThings/matter-lock/src/test/test_matter_lock_cota_no_lock_codes_profile.lua

```diff
@@ -0,0 +1,331 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+-- Mock out globals
+local test = require "integration_test"
+local capabilities = require "st.capabilities"
+test.add_package_capability("lockAlarm.yml")
+local t_utils = require "integration_test.utils"
+local clusters = require "st.matter.clusters"
+local DoorLock = clusters.DoorLock
+
+local mock_device_record = {
+  profile = t_utils.get_profile_definition("lock-lockalarm-nobattery.yml"),
```

Comment:
Since this is the same profile as used in `test_aqara_matter_lock`, can we swap in `lock-nobattery-notamper` and have the tests still pass?

Reply:
I think I am getting at the same thing as others wrt the unit tests, forgot to refresh before my review.

Reply:
Thanks Carter! I made that change to the unit test in [7f31271](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1492/commits/7f3127131083e1d9bb99689e68f460451e517ee5), so now the test uses the `lock-nocodes-notamper` profile, which does not have the `lockCodes` capability, while the `base-lock` profile that does have the `lockCodes` capability is used in the complimentary `test_matter_lock_cota.lua` test file. So these two files should test COTA for both device that have and devices that don't have `lockCodes` in their profiles.